src/kbmod/analysis_utils.py

```diff
@@ -90,25 +93,40 @@ def load_images(
             # Compute the full file path for loading.
             full_file_path = os.path.join(im_filepath, visit_file)
 
-            # Load the image info from the FITS header.
-            header_info = ImageInfo()
-            header_info.populate_from_fits_file(full_file_path)
+            # Try loading information from the FITS header.
+            visit_id = None
+            with fits.open(full_file_path) as hdu_list:
+                curr_wcs = WCS(hdu_list[1].header)
+
+                # If the visit ID is in header (using Rubin tags), use for the visit ID.
+                # Otherwise extract it from the filename.
+                if "IDNUM" in hdu_list[0].header:
+                    visit_id = str(hdu_list[0].header["IDNUM"])
+                else:
+                    name = full_file_path.rsplit("/")[-1]
```

Comment:
if it's a path, better `os.path.split`, it accounts for escapes and os defined separators

Reply:
Done.